### PR TITLE
Avatar: rewrite with a default colored avatar

### PR DESCRIPTION
**Pull request by G-Ray**

### CODE REVIEW

**Review by leventdem — APPROVED** [review]
ok, tested.

File: src/components/Avatar/Avatar.jsx [hunk 1]
```diff
@@ -1,33 +1,20 @@
...
-  onClick: PropTypes.func
-}
+const COLORS = [
+  styles.colorPrimary,
+  styles.colorSuccess,
+  styles.colorDanger,
+  styles.colorNeutral,
+  styles.colorWarning,
+  styles.colorCyan
+]
```
Comment (sujeebant) [inline src/components/Avatar/Avatar.jsx:17]: I think the colors used should be redefined with different names in colors.scss.
It seems strange to use colors declared as colorSuccess or colorDanger.

File: src/components/Avatar/Avatar.jsx [hunk 2]
```diff
@@ -40,25 +27,39 @@ export default class Avatar extends React.Component {
...
-            accept='.jpg, .jpeg, .png'
-          />
+    return (
+      <div
+        style={{ width: size, height: size }}
+        onClick={upload ? this.openDialog : null}
+        className={classNames(
+          styles.Avatar,
+          { [styles.upload]: upload && (image || username) },
+          { [styles.new]: !image && !username },
```
Comment (sujeebant) [inline src/components/Avatar/Avatar.jsx:40]: Isn't there supposed to always have a username that is not undefined ?

File: src/components/Avatar/Avatar.jsx [hunk 3]
```diff
@@ -40,25 +27,39 @@ export default class Avatar extends React.Component {
...
+
+        <Camera className={styles.camera} size={42} />
 
-          {image
-            ? <Replace size={size} image={image} onClick={this.openDialog} />
-            : <NewUpload style={style} onClick={this.openDialog} />
-          }
-        </div>)
-    }
+        {!image && username && (
```
Comment (sujeebant) [inline src/components/Avatar/Avatar.jsx:51]: same here
(Isn't there supposed to always have a username that is not undefined ?)

File: src/components/Avatar/Avatar.jsx (hunk 2 above)
Comment (G-Ray) [reply src/components/Avatar/Avatar.jsx:40]: there is the case where we have not entered any username yet, when signing up, that's the only case I think

Comment (sujeebant) [reply src/components/Avatar/Avatar.jsx:40]: okay ! I didn't think of that case !

**Pull request merged**